src/main/java/faang/school/godbless/error_prettifier/PrettyError.java

```diff
@@ -0,0 +1,14 @@
+package faang.school.godbless.error_prettifier;
+
+import java.util.function.Supplier;
+
+public class PrettyError{
+    public static <T> T withErrorHandling(Supplier<T> supplier, ExceptionHandler exceptionHandler) {
+        try {
+            return supplier.get();
+        } catch (RuntimeException e) {
+            exceptionHandler.handle(e);
+            throw e;
```

Comment:
like this?

Reply:
Not really. You don't have to throw anything, you can just return exceptionHandler.handle(e)